### ISSUE TITLE
Can't get the lastest version (v0.2.2) from cargo

### ISSUE THREAD

**Issue by quoctuan0405** [luketpeterson/fuzzy_rocks#2]
## Describe the issue
I can't get the lastest version of fuzzy_rocks (v0.2.2) from cargo. I add the following line in cargo.toml:

```
[dependencies]
fuzzy_rocks = "0.2.2"
```
but when run `cargo run`, I got back the following error:
```
error: failed to select a version for the requirement `fuzzy_rocks = "^0.2.2"`
candidate versions found which didn't match: 0.2.1, 0.1.1
location searched: crates.io index
required by package `fuzzy-rocks-sandbox v0.1.0 (D:\rust\sandbox\rocksdb)`
perhaps a crate was updated and forgotten to be re-vendored?
```

_New version 0.2.2 with updated dependency rocksdb v0.18.0_

![ausdhofaisdfd-edit](https://github.com/luketpeterson/fuzzy_rocks/assets/29013036/654ea4f8-ae55-4703-a97a-30ad8f882667)

## Expected behaviour

Get the lastest version (0.2.2) from cargo. 

Hope you read this issue and appreciate your help :flushed::flushed::flushed::pray::pray:

**Comment by quoctuan0405** [comment]
fuzzy_rocks v0.2.1 use rocksdb v0.16.0 which somehow I can't build on my Window machine. I already have LLVM and gcc installed.

```
error: failed to run custom build command for `librocksdb-sys v6.20.3`

Caused by:
  process didn't exit successfully: `D:\rust\sandbox\rocksdb\target\debug\build\librocksdb-sys-8df16fa9fce77774\build-script-build` (exit code: 101)
  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: LexError { span: Span }', D:\Program Files\rust\registry\src\index.crates.io-6f17d22bba15001f\bindgen-0.59.2\src\codegen\mod.rs:4577:66
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

But rocksdb v.0.18.0 work just fine. With the example in rust-rocksdb docs (https://docs.rs/rocksdb/latest/rocksdb): 

```
use rocksdb::{DB, Options};
// NB: db is automatically closed at end of lifetime
let path = "_path_for_rocksdb_storage";
{
   let db = DB::open_default(path).unwrap();
   db.put(b"my key", b"my value").unwrap();
   match db.get(b"my key") {
       Ok(Some(value)) => println!("retrieved value {}", String::from_utf8(value).unwrap()),
       Ok(None) => println!("value not found"),
       Err(e) => println!("operational problem encountered: {}", e),
   }
   db.delete(b"my key").unwrap();
}
let _ = DB::destroy(&Options::default(), path);
```

I got back:

```
retrieved value: my value
```

## System information

- OS Name: Microsoft Windows 10 Pro
- Version: 10.0.19045 Build 19045
- Clang version: 16.0.2 (Target: x86_64-pc-windows-msvc, Thread model: posix)
- GCC version: 12.2.0 (Rev10, Built by MSYS2 project)

**Comment by luketpeterson** [comment]
I don't understand the issue.  fuzzy_rocks 0.2.2 should work with any version of rocksdb v. 0.18.0 or newer.

have you run `cargo update` to fetch the latest version?

I don't have access to a Windows 10 machine, but if you would like me to roll back to rocksdb v. 0.16.0 as a minimum requirement, I *think* that will be fine.

**Comment by quoctuan0405** [comment]
When i run `cargo update` I get the lastest version of fuzzy_rocks which is 0.2.1 but not 0.2.2.

> I don't have access to a Windows 10 machine, but if you would like me to roll back to rocksdb v. 0.16.0 as a minimum requirement, I think that will be fine.

No I want you to update it to newer version of rocksdb, not backward.

**Comment by luketpeterson** [comment]
Ok.  I misunderstood.  I thought you were saying the minimum version of rocksdb was too new.

Did you try running `cargo vendor`?

**Comment by luketpeterson** [comment]
Oh.... Sorry.  I failed to notice you were talking about an unpublished version.  I just published it.  Try now.

Although I realize it has a warning with the latest tools, so I'll publish a new version with that fixed shortly.

**Comment by quoctuan0405** [comment]
Ok I can get the lastest version now. Thanks :smile::smile::smile:

Appreciate your help :pray:

**Comment by luketpeterson** [comment]
@quoctuan0405 I just published v0.2.3 which has the warning fixed and another bug fix as well.

Thanks for reporting the issue.

**Comment by quoctuan0405** [comment]
Thanks @luketpeterson :smile:
